src/ToolBox/SOS/Strike/eeheap.cpp

```diff
@@ -880,12 +879,30 @@ BOOL GetSizeEfficient(DWORD_PTR dwAddrCurrObj,
         // from the target
         DacpMethodTableData dmtd;
         // see code:ClrDataAccess::RequestMethodTableData for details
-        if (dmtd.Request(g_sos,dwAddrMethTable) != S_OK)
-            return FALSE;
+        if (dmtd.Request(g_sos, dwAddrMethTable) != S_OK)
+            return NULL;
+
+        DacpMethodTableCollectibleData dmtcd;
+        if (dmtcd.Request(g_sos, dwAddrMethTable) != S_OK)
```

Comment:
Eventually this will need to be ported to the diagnostics repo in which SOS can run against past, present and future versions of the runtime.  So this Request call for the new DAC interface should fail gracefully and not the whole gc command.  I guess you can assume that the type isn't collectible if the call fails.

Reply:
Thanks Mike for the comment, it makes sense.